islandora_transforms/BPL_MODS_to_solr.xslt

```diff
@@ -0,0 +1,130 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- BPL requires some indexing of cases not handled by standard MODS slurping -->
+<xsl:stylesheet version="1.0"
+  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+  xmlns:foxml="info:fedora/fedora-system:def/foxml#"
+  xmlns:mods="http://www.loc.gov/mods/v3"
+  exclude-result-prefixes="mods">
+
+  <!--Non-date-typed regular name/namePart -->
+  <xsl:template mode="slurp_for_bpl" match="mods:name[@type='personal']/mods:namePart[not(@type='date')]">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'name_personal_namePart_not_date'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Non-date-typed subject name/namePart -->
+  <xsl:template mode="slurp_for_bpl" match="mods:subject/mods:name[@type='personal']/mods:namePart[not(@type='date')]">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'subject_name_personal_namePart_not_date'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Make sure subject/name/nameParts of all types are being indexed (no
+       non-typed field is currently indexed for subject/name) -->
+  <xsl:template mode="slurp_for_bpl" match="mods:subject/mods:name/mods:namePart">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'subject_name_namePart_not_typed'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Lump faceted subject fields together -->
+  <xsl:template mode="slurp_for_bpl" match="mods:subject/mods:topic | mods:subject/mods:name[@type='personal']/mods:namePart[not(@type='date')]">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'faceted_subjects'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Lump together all descendants of subject for advanced search -->
+  <xsl:template mode="slurp_for_bpl" match="mods:subject//*">
```

Comment:
A copy of writing `all_subject_descendants` has been moved to all the other `mods:subject` fields